### DISCUSSION TITLE
1.0 proposal for environments

### DISCUSSION THREAD

**Discussion by gakesson** [helmfile/helmfile#292] [Questions] [answered]
In our project we rely upon the environments section to be rendered, in order not to hardcode the available environments in various testing pipelines (consumer is responsible for defining its own helmfile environment). I.e. we produce and ship the helmfile.yaml but not the environments.

Example on how environments are rendered in the helmfile.yaml:

**helmfile.yaml**
```
{{ $configDir := default (env "CONFIG_DIR") | default "config" }}
{{ tpl (readFile (printf "%s/envs.yaml" $configDir)) . }}
---
repositories:
- name: incubator
  url: https://charts.helm.sh/incubator
---
releases:
- name: service-a
  chart: incubator/raw
```

**config/envs.yaml**
```
environments:
  perf:
    values:
      - some-key: some-value

```

This allows me to run `helmfile -e perf list`

Reading the point 1 in [Towards Helmfile 1.0](https://github.com/helmfile/helmfile/blob/main/docs/proposals/towards-1.0.md) it is not clear to me if the proposal will still support this type of environments section rendering that we're using? I would appreciate if someone (perhaps @mumoshu that wrote the doc?) could elaborate on this.

**Comment by mumoshu** [comment, marked as answer]
@gakesson Hey! Thanks for reviewing the proposal. Re your question, it would just work, despite that you'd need to rename `helmfile.yaml` to `helmfile.yaml.gotmpl` or `helmfile.yaml.tpl`.

The topic in the proposal is about the case where you didn't have `---` after the environments section. You do have `---`s across sections so it's behavior is well-defined and will require no changes in helmfile 1.0.

**Reply by gakesson** [reply]
Very good, thanks for quick reply! 
Renaming of helmfile.yaml we can manage (in fact we can adapt to all 3 points without much effort).
